sw/device/silicon_creator/mask_rom/mask_rom_start.S

```diff
@@ -132,6 +132,16 @@ _mask_rom_start_boot:
    * need to be careful.
    */
 
+  // Check if we should halt here.
+  li   a0, (TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR + \
+            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET)
+  lw   t0, OTP_CTRL_PARAM_CREATOR_SW_CFG_ROM_EXEC_EN_OFFSET(a0)
+  bnez t0, .L_exec_en
+.L_halt_loop:
+  wfi
+  j .L_halt_loop
```

Comment:
We can also have a busy-poll like construct that repeatedly reads from the OTP but IIRC we thought `wfi` could be better since we won't be trashing the bus.

@cfrantz @tjaychen @moidx for visibility.